File: ClientApp/src/app/dto/clothes.dto.ts
import { GenderType } from "./gender-type";
import { Id } from "../models/id";
import { ClothesSize } from "./clothes-size";

export interface IClothesDto {
  clothesId: Id;
  name: string;
  price: number;
  pictureUrl: string;
  description: string;
  color: string;
  genderType: GenderType;
  size: ClothesSize;
  brandId: Id;
  isOrdered: boolean;
}

File: Server/Dto/ClothesDto.cs
using System.ComponentModel.DataAnnotations;

namespace Server.DTO
{
    public class ClothesDto
    {
        public int ClothesId { get; set; }
        
        [Required]
        public string Name { get; set; }
        
        public string Description { get; set; }

        [Required]
        public decimal Price { get; set; }
        
        public string PictureUrl { get; set; }
        
        public string Color { get; set; }

        [Required]
        public ushort GenderType { get; set; }
        
        [Required]
        public ushort Size { get; set; }
        
        public bool IsOrdered { get; set; }
        
        [Required]
        public int BrandId { get; set; }
    }
}